### PR TITLE
AoC 2020 commands

### LINKED ISSUE #162: commands for AoC 2020
leaderboard command and join command for code collective server leader board

### PR DESCRIPTION
**Pull request by saiTama-max**
Added  2 commands for Code Collective server for AoC 2020, join (to join the Code Collective leaderboard) and leaderboard (to view the leaderboard)
closes #162

### CODE REVIEW

**Comment by fisher60** [comment]
@saiTama-max do we want multiple servers to be able to join our personal leaderboard though?

**Comment by saiTama-max** [comment]
@fisher60 i was thinking about that, in some cases it could be good for more competition, but if the server doesn't agree then i could remove it

File: bot/cogs/advent_of_code.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        for ms in stat_d['members']:
+            if stat_d['members'][ms]['name'] is None:
+                stat_d['members'][ms]['name'] = 'Anonymous'
+            stats.update({(stat_d['members'][ms]['name'],
+                         stat_d['members'][ms]['stars']): stat_d['members'][ms]['local_score']})
+
+        stats = {k: stats[k] for k in sorted(stats, key=lambda y: stats[y])[::-1]}
+        return stats
+
+    @group(name="adventofcode", aliases=('aoc', 'aoc2020'))
```
Comment (fisher60) [inline bot/cogs/advent_of_code.py:29]: missing `brief`, causes help command to error

File: bot/cogs/advent_of_code.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    async def aoc_group(self, ctx: Context) -> None:
+        """Group for advent of code commands."""
+        if not ctx.invoked_subcommand:
+            await ctx.send("Please enter a valid command")
+
+    @aoc_group.command(brief="Get the leaderboard join code in your DM's",
+                       usage=".join",
```
Comment (fisher60) [inline bot/cogs/advent_of_code.py:36]: usage incorrect, correct usage is `.aoc join`

File: bot/cogs/advent_of_code.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        error_msg = f":x: {ctx.author.mention}, please (temporarily) enable DMs to receive the join code"
+
+        await ctx.message.add_reaction("📨")
+        try:
+            await ctx.author.send('\n'.join(info))
+        except errors.Forbidden:
+            await ctx.send(error_msg)
+
+    @aoc_group.command(brief="Get the leaderboard of AoC 2020 for this server",
+                       usage=".leaderboard",
```
Comment (fisher60) [inline bot/cogs/advent_of_code.py:54]: usage should be `.aoc leaderboard`

**Review by fisher60 — CHANGES_REQUESTED** [review]
overall this looks great. just some quick, small fixes to finish it up.

**Comment by saiTama-max** [comment]
uh i just realized theres a bug in the rendering, its mismatched when the star cont is 2 digit now, i will fix that soon

**Review by fisher60 — APPROVED** [review]
This looks great, runs perfect, ty.

**Pull request merged**